meterfs/meterfs.c

```diff
@@ -382,8 +383,15 @@ int meterfs_writeRecord(file_t *f, const void *buff, size_t bufflen, meterfs_ctx
 	e.id.no = f->lastidx.no + 1;
 	e.id.nvalid = 0;
 
-	ctx->write(ctx->offset + f->header.sector * ctx->sectorsz + offset + sizeof(entry_t), (void *)buff, bufflen);
-	ctx->write(ctx->offset + f->header.sector * ctx->sectorsz + offset, &e, sizeof(entry_t));
+	stat = ctx->write(ctx->offset + f->header.sector * ctx->sectorsz + offset + sizeof(entry_t), (void *)buff, bufflen);
+	if (stat < 0)
+		return stat;
+	else
+		wrote = stat;
```

Comment:
if ((wrote = ctx->write(ctx->offset + f->header.sector * ctx->sectorsz + offset + sizeof(entry_t), (void *)buff, bufflen)) < 0)
return wrote;

is somewhat more compact form

Reply:
done